### PR TITLE
Release v0.2.0

### PR DESCRIPTION
**Pull request by inancgumus**
This adds release notes and changes to the README for the new release.

### CODE REVIEW

File: release notes/v0.2.0.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+This release adds three new features:
+* HTTP Basic Auth support.
+* Automatically adjusting viewport insets depending on the operating system.
+* Denying IPs and hostnames.
+
+Fixes bugs, including a major one in `WaitForFrameNavigation,` and we now handle the extension shutting-down stage more robustly. Improves testability, logging, and user error messages.
```
Comment (imiric) [inline release notes/v0.2.0.md:6]: I wouldn't want to highlight the `WaitForFrameNavigation` fix; it's only relevant for dynamic sites, and we want them to be using `waitForSelector()` anyway.

```suggestion
In addition, it comes with a few important bug fixes, and we now handle the extension shutting-down stage more robustly. It also improves testability, logging, and user error messages.
```

File: release notes/v0.2.0.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+This release adds three new features:
+* HTTP Basic Auth support.
+* Automatically adjusting viewport insets depending on the operating system.
+* Denying IPs and hostnames.
```
Comment (imiric) [inline release notes/v0.2.0.md:4]:
```suggestion
* Blocking IPs and hostnames.
```

File: release notes/v0.2.0.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  We short-circuit the browser communication when the connection is closed. However, sometimes, we may do so when the browser closes or the k6 runner abrupts the test run. This fix is for such situations and helps properly short-circuit the communication when it should end.
+
+- Fix flaky ElementHandle click with detached node. ([#219](https://github.com/grafana/xk6-browser/pull/219))
+
+  Clicking on an element the second time created problems if the first one resulted in an exception.
+
+- Fix Browser.onAttachedToTarget nil pointer. ([#217](https://github.com/grafana/xk6-browser/pull/217))
+
+  There was a race condition in `Connection` and `FrameSession` when the connection resulted in an error. This change removes running browser attachment and detachment concurrently.
```
[the remark this answers is not in the document]
Comment (inancgumus) [reply release notes/v0.2.0.md:60]: Yeah, but think about future contributors and ourselves. They are useful info.

[the remark this answers is not in the document]
Comment (imiric) [reply release notes/v0.2.0.md:60]: Any contributor can just look at the PRs and Git log if they want further details.

This is why it's a bad idea to generate changelogs from `git log`. Changelogs should be written manually and from the perspective of the user, not the developer.

**Comment by imiric** [comment]
@inancgumus See the changes I just pushed and let me know. All PRs since v0.1.3 are mentioned, except trivial things like README, CI and example changes, and reversion PRs.

@olegbespalov If you don't mind giving it a quick look, we'd appreciate it! :pray:

File: release notes/v0.2.0.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  This has the same effect as passing the [`host-resolver-rules` argument](https://datacadamia.com/web/browser/chrome#dns_resolver) to Chrome via [browser args](https://github.com/grafana/xk6-browser/blob/ae264530b6fbe0285ac9df9b0f4a907b460bb805/examples/browser_args.js).
+
+
+## Bugs fixed
+
+- `Page.queryAll()` and `Element.queryAll()` not finding elements. ([#195](https://github.com/grafana/xk6-browser/pull/195))
+
+- Possible issue with consecutive clicks on an element if the first click resulted in an exception. ([#219](https://github.com/grafana/xk6-browser/pull/219))
+
+- We now handle the extension shutting-down stage more robustly, and it shouldn't print any errors. ([#188](https://github.com/grafana/xk6-browser/pull/188), )
```
Comment (olegbespalov) [inline release notes/v0.2.0.md:59]:
```suggestion
- We now handle the extension shutting-down stage more robustly, and it shouldn't print any errors. ([#188](https://github.com/grafana/xk6-browser/pull/188))
```

**Review by olegbespalov — APPROVED** [review]
:+1:

File: release notes/v0.2.0.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+xk6-browser v0.2.0 is here! :tada:
+
+This is a minor release with a few important features, several bug fixes, stability improvements and overall code cleanup.
```
Comment (inancgumus) [inline release notes/v0.2.0.md:3]:
```suggestion
This is a minor release with a few new features, several bug fixes, stability improvements, and overall code cleanup.
```

File: release notes/v0.2.0.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  ```javascript
+  // Correct
+  const context = browser.newContext({
+    httpCredentials: {
+      username: 'user',
+      password: 'pass',
+    }
+  });
+
+  // The following is being deprecated
```
Comment (inancgumus) [inline release notes/v0.2.0.md:25]:
```suggestion
  // The following is deprecated
```

File: release notes/v0.2.0.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+  Browser insets change depending on the operating system, and this feature adds support for adjusting insets of the viewport for Windows, Linux, and macOS.
+
+  Note that it won't update the viewport if the browser runs in headless mode.
+
+  See the related issues for more details: [#182](https://github.com/grafana/xk6-browser/issues/182), [#183](https://github.com/grafana/xk6-browser/issues/183).
+
+- Blocking requests based on the remote host's name or IP via k6's [`blacklistIPs`](https://k6.io/docs/using-k6/options/#blacklist-ips) and [`blockHostnames`](https://k6.io/docs/using-k6/options/#block-hostnames) options. ([#204](https://github.com/grafana/xk6-browser/pull/204), [#205](https://github.com/grafana/xk6-browser/pull/205))
+
+  If the script specifies these options, the browser will block matching requests with an `ERR_BLOCKED_BY_CLIENT` error.
```
Comment (inancgumus) [inline release notes/v0.2.0.md:46]:
```suggestion
  The browser will block matching requests with an `ERR_BLOCKED_BY_CLIENT` error if the script specifies these options.
```

File: release notes/v0.2.0.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  This has the same effect as passing the [`host-resolver-rules` argument](https://datacadamia.com/web/browser/chrome#dns_resolver) to Chrome via [browser args](https://github.com/grafana/xk6-browser/blob/ae264530b6fbe0285ac9df9b0f4a907b460bb805/examples/browser_args.js).
+
+
+## Bugs fixed
+
+- `Page.queryAll()` and `Element.queryAll()` not finding elements. ([#195](https://github.com/grafana/xk6-browser/pull/195))
+
+- Possible issue with consecutive clicks on an element if the first click resulted in an exception. ([#219](https://github.com/grafana/xk6-browser/pull/219))
+
+- We now handle the extension shutting-down stage more robustly, and it shouldn't print any errors. ([#188](https://github.com/grafana/xk6-browser/pull/188))
```
Comment (inancgumus) [inline release notes/v0.2.0.md:59]:
```suggestion
- `Page.waitForNavigation` becomes unresponsive if the browser or connection is closed or a user cancels the script. We now handle the extension shutting-down stage more robustly, and it shouldn't become unresponsive. ([#188](https://github.com/grafana/xk6-browser/pull/188))
```

File: release notes/v0.2.0.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+- Passing arguments to the browser command now works correctly; see [this example](https://github.com/grafana/xk6-browser/blob/ae264530b6fbe0285ac9df9b0f4a907b460bb805/examples/browser_args.js). ([#206](https://github.com/grafana/xk6-browser/pull/206))
+
+- `Page.waitForNavigation()` doesn't timeout anymore for navigation within the same document (e.g. via anchor links or the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API)). ([#247](https://github.com/grafana/xk6-browser/pull/247)).
+
+- Several internal fixes that improve stability: [#188](https://github.com/grafana/xk6-browser/pull/188), [#217](https://github.com/grafana/xk6-browser/pull/217), [#238](https://github.com/grafana/xk6-browser/pull/238).
```
Comment (inancgumus) [inline release notes/v0.2.0.md:65]: We explained #188 before. Should we repeat it here?

```suggestion
- Internal fixes that improve stability: [#217](https://github.com/grafana/xk6-browser/pull/217) and [#238](https://github.com/grafana/xk6-browser/pull/238).
```

**Review by inancgumus — COMMENTED** [review]
Much better than before, good job 👏

**Comment by inancgumus** [comment]
LGTM, should we merge it? @imiric

**Comment by imiric** [comment]
In about an hour, when we're ready to release. I'll just clean up the history here a bit before that.

**Pull request merged**